mlir/lib/Dialect/Vector/IR/VectorOps.cpp

```diff
@@ -5184,6 +5184,23 @@ std::optional<SmallVector<int64_t, 4>> GatherOp::getShapeForUnroll() {
   return llvm::to_vector<4>(getVectorType().getShape());
 }
 
+/// Cheeck if `indexVec` is constant 1D vec of consecutive values [0, 1, 2, ...]
+static LogicalResult isContiguousIndices(Value indexVec) {
```

Comment:
This method doesn't care where the input came from (it could be indices, but it could be sth else). I suggest generalising the name.
```suggestion
static LogicalResult isZeroBasedContiguousSeq(Value indexVec) {
```